### DISCUSSION TITLE
feature request: download zip artefacts as .zip, not .zip.zip

### DISCUSSION THREAD

**Discussion by tombrus** [community/community#26223] [Actions]
<p>After a action is finished I can download the produced artifacts from a popup menu in the upper right corner of the build log.</p>
<p>When it is a zip file it will show up as " <strong>a-file.zip</strong>" but after downloading it will show up as " <strong>a-file.zip.zip</strong>" on my disk and I need to unpack twice.</p>
<p>Can the download logic be changed so that zip files are not re-zipped?</p>
<p>Thanks!</p>
<p>Tom</p>

**Comment by BrightRan** [comment]
<p>Did you use the action <a href="https://github.com/actions/upload-artifact/tree/v1" rel="nofollow noopener"><strong>actions/upload-artifact</strong></a> to upload artifacts in your workflow?</p>
<p>I tested and checked this with different OSs, all the downloaded artifacts were normal and shown as .zip, not .zip.zip .</p>
<p>If possible, can you share your repository with us, so that we can check your workflow configurations to analyze the root cause?</p>

**Comment by tombrus** [comment]
<p>Thanks for the quick reply BrightRan!</p>
<p>On <a href="https://github.com/ModelingValueGroup/DclareForMPS/actions/runs/36896746" rel="nofollow noopener">this page</a> there is a <strong><a href="https://github.com/ModelingValueGroup/DclareForMPS/suites/447834216/artifacts/1710137" rel="nofollow noopener">DclareForMPS.zip</a></strong> that downloads on my mac (safari 13.0.5) as  <strong>DclareForMPS.zip.zip</strong>.</p>
<p>I used  <strong>actions/upload-artifact@v1</strong>.</p>
<p>You will find whatever you need in the underlaying project.</p>
<p>Hope you can shed some light :slight_smile:</p>
<p>-Tom</p>

**Comment by BrightRan** [comment, marked as answer]
<p>When using the action  <strong>actions/upload-artifact</strong> to upload artifacts, the action will automatically package the files in the specific <strong>path</strong>  (can be a directory path or a file path) as a ZIP and names the ZIP with the specific <strong>name</strong> (artifact name).</p>
<p>I checked your workflow configurations, and found:</p>
<pre><code>- name: "publish action-artifact zip"
        uses: actions/upload-artifact@v1
        with:
          name: DclareForMPS.zip
          path: build/artifacts/DclareForMPS/DclareForMPS.zip
</code></pre>
<p>There are 2 questions in your workflow about this:</p>
<ol>
<li>
<p>You have set the artifact name as <strong>DclareForMPS.zip</strong> , so the artifact will be zipped as <strong>DclareForMPS.zip.zip.</strong></p>
</li>
<li>
<p>Before upload the artifact, you have zipped the files as a ZIP in the specific <strong>path</strong> , that means the uploaded file itself is already a ZIP file. This is the root cause that you must unzip your artifact twice.</p>
</li>
</ol>
<p>After executing the action  <strong>actions/upload-artifact</strong> , the structure of your artifact will like below:</p>
<pre><code>DclareForMPS.zip.zip
┕ DclareForMPS.zip
</code></pre>
<p>To avoid this misunderstanding, I recommend you take the following updates:</p>
<ol>
<li>
<p>Set the artifact name as <strong>DclareForMPS</strong> , not  <strong>DclareForMPS.zip</strong>.</p>
</li>
<li>
<p>Before upload the artifact, directly copy the folders or files you want to upload as artifact into the specific <strong>path</strong>  ( <strong>build/artifacts/DclareForMPS</strong> ), instead of zipping them at first.</p>
</li>
<li>
<p>After above two updates, the step configuration should look like below:</p>
<ul>
<li>name: “publish action-artifact zip”<br>
uses: actions/upload-artifact@v1<br>
with:<br>
name: DclareForMPS<br>
path: build/artifacts/DclareForMPS</li>
</ul>
</li>
</ol>

**Reply by brianjmurrell** [reply]
I think the real root problem of this issue is the inability to download artifacts in any other format than having them zipped before being sent.  This is really super annoying to be completely honest about it.  If I want to peruse a number of logfiles in artifacts for example, every one I click on first requires me to jump through the extract-archive process.

**Comment by tombrus** [comment]
<p>That explains a lot! Thanks.</p>
<p>I think I was a bit misled by the term <em>artifact</em> which sounds more like a file then a dir to me, but that is bias ;-).</p>
<p>This gives me the workaround I needed.</p>
<p>I would still like to propose this as a feature request though:</p>
<p>   - do not zip an already zipped file if it is the only artifact</p>
<p>Thanks again!</p>

**Comment by rusefillc** [comment]
Hello from 2024!

Please consider not zipping zip files exactly as requested above.

**Comment by mkorsukov** [comment]
It would be nice having `skipUnzip` boolean option for the `actions/download-artifact@v4` to get the artifact(s) as is (zip-compressed). I do my best to name the artifact the desired way (eg. AppName-version+hash.zip) for `actions/upload-artifact@v4` in the previous job / step, and I need to simply get it later.

I think this is related issue: https://github.com/actions/upload-artifact/issues/405

**Comment by flywire** [comment]
See also https://github.com/actions/upload-artifact/issues/426
